### PR TITLE
Various GL fixes

### PR DESCRIPTION
**Pull request by omni-viral**
Fixes number of small issues I encountered

### CODE REVIEW

File: src/backend/gl/src/info.rs [hunk 1]
```diff
@@ -140,6 +140,15 @@ fn get_usize(gl: &GlContainer, name: u32) -> Result<usize, Error> {
         Ok(value as usize)
     }
 }
+fn get_u64(gl: &GlContainer, name: u32) -> Result<u64, Error> {
```
Comment (kvark) [inline src/backend/gl/src/info.rs]: why is this needed if we have `get_usize` implemented the same way?

File: src/backend/gl/src/info.rs [hunk 2]
```diff
@@ -347,9 +356,9 @@ pub(crate) fn query_all(gl: &GlContainer) -> (Info, Features, LegacyFeatures, Li
         max_viewports: 1,
         optimal_buffer_copy_offset_alignment: 1,
         optimal_buffer_copy_pitch_alignment: 1,
-        min_texel_buffer_offset_alignment: 1,   // TODO
-        min_uniform_buffer_offset_alignment: 1, // TODO
-        min_storage_buffer_offset_alignment: 1, // TODO
+        min_texel_buffer_offset_alignment: 1,
+        min_uniform_buffer_offset_alignment: get_u64(gl, glow::UNIFORM_BUFFER_OFFSET_ALIGNMENT).unwrap_or(1024),
```
Comment (kvark) [inline src/backend/gl/src/info.rs]: great!

File: src/backend/gl/src/lib.rs [hunk 3]
```diff
@@ -655,7 +655,17 @@ impl hal::PhysicalDevice<Backend> for PhysicalDevice {
         _usage: image::Usage,
         _view_caps: image::ViewCapabilities,
     ) -> Option<image::FormatProperties> {
-        None //TODO
+        Some(image::FormatProperties {
+            max_extent: image::Extent {
+                width: !0,
```
Comment (kvark) [inline src/backend/gl/src/lib.rs]: let's at least leave a TODO to return the actual limits

File: src/backend/gl/src/window/web.rs [hunk 4]
```diff
@@ -153,7 +153,9 @@ impl Device {
 
         let (int_format, iformat, itype) = match config.format {
             f::Format::Rgba8Unorm => (glow::RGBA8, glow::RGBA, glow::UNSIGNED_BYTE),
+            f::Format::Bgra8Unorm => (glow::RGBA8, glow::BGRA, glow::UNSIGNED_BYTE),
             f::Format::Rgba8Srgb => (glow::SRGB8_ALPHA8, glow::RGBA, glow::UNSIGNED_BYTE),
+            f::Format::Bgra8Srgb => (glow::SRGB8_ALPHA8, glow::BGRA, glow::UNSIGNED_BYTE),
```
Comment (kvark) [inline src/backend/gl/src/window/web.rs]: do we have to repeat the code from `create_swapchain` here?

Comment (omni-viral) [reply src/backend/gl/src/window/web.rs]: `create_swapchain` under `#[cfg(target_arch = "wasm32")]` just calls this function.

File: src/backend/gl/src/device.rs [hunk 5]
```diff
@@ -842,17 +842,17 @@ impl d::Device<B> for Device {
...
-                // Sampler2D won't show up in UniformLocation and the only other uniforms
-                // should be push constants
-                uniforms.push(n::UniformDesc {
-                    location: location as _,
-                    offset,
-                    utype,
-                });
-
-                offset += size as u32;
+                if let Some(location) = gl.get_uniform_location(program, &name) {
```
[the remark this answers is not in the document]
Comment (omni-viral) [reply src/backend/gl/src/device.rs]: I just made the code to do the same it was doing before porting to glow.

[the remark this answers is not in the document]
Comment (kvark) [reply src/backend/gl/src/device.rs]: ok, fair enough

**Comment by kvark** [comment]
Ok, I think we can go ahead with this. The rough parts are what we already have, and the PR doesn't regress them.
bors r+
